src/TextInput/TextInput.tsx

```diff
@@ -144,7 +144,7 @@ const TextInput = React.forwardRef<HTMLInputElement, TextInputProps>(
           onFocus={handleInputFocus}
           onBlur={handleInputBlur}
           type={type}
-          aria-required={required ? 'true' : 'false'}
+          aria-required={required}
```

Comment:
Just a quick question do we want to have `aria-required="false"` if required is passed but as false? (Example [usage](https://github.com/github/internal-statuspage/blob/30b1e7bc7a53daeec20e4c802578007e3654f9ae/frontend/src/incidents/DeclareIncidentDialog/DeclareIncidentForm.tsx#L82C9-L82C25))

Or do we only want to have `aria-required` when it is only true?

Reply:
> Or do we only want to have aria-required when it is only true?

yes :)

This is the failing test: https://github.com/github/github/actions/runs/7202649701/job/19621310998?pr=304951